sw/device/tests/dif/dif_uart_sanitytest.c

```diff
@@ -0,0 +1,70 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_uart.h"
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/runtime/hart.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"  // Generated.
+
+extern bool uart_reconfigure_required;
+
+static dif_uart_t uart0;
+static const uint8_t kSendData[] = "Sanity test!";
+
+static bool uart_initialise(mmio_region_t base_addr, dif_uart_t *uart) {
+  dif_uart_config_t config = {
+      .baudrate = kUartBaudrate,
+      .clk_freq_hz = kClockFreqHz,
+      .parity_enable = kDifUartDisable,
+      .parity = kDifUartParityEven,
+  };
+
+  if (dif_uart_init(base_addr, &config, uart) != kDifUartConfigOk) {
+    return false;
+  }
+
+  return true;
```

Comment:
Get rid of the `if` and return the `!=` expression directly.

Since this is a test though, I would personally wrap it in a CHECK (like we've been doing in DV tests). CHECK tends to be better for tests, since it captures a line/file information, making it easier to find failures without breaking out a debugger.

Reply:
`CHECK` uses UART underneath to print out messages, as we are testing UART here, I wanted to avoid printing things altogether. But generally I agree, I used `CHECK` in my pinmux/gpio/padctrl sanity test.

Reply:
Missed the first time around, but fixed now:

> Get rid of the if and return the != expression directly.

Reply:
CHECK just uses whatever the current stdout sink is. You can disable it with `base_set_stdout(NULL)` at the start of main.

Might be worth it writing a SPI output sink for this kind of test at some point... *shrug*.

Reply:
> You can disable it with base_set_stdout(NULL) at the start of main.

I think there might be problem with this, as CHECK would internally abort if condition is not met, and while sink is `NULL` it then will not print "FAIL", which is needed for CI. That will make the test time-out eventually and fail, but probably increase the runtime.